src/main/scala/com/databricks/labs/mosaic/gdal/MosaicGDAL.scala

```diff
@@ -22,17 +23,36 @@ object MosaicGDAL extends Logging {
 
     def prepareEnvironment(spark: SparkSession, initScriptPath: String, sharedObjectsPath: String): Unit = {
         if (!wasEnabled(spark) && !isEnabled) {
-            copyInitScript(initScriptPath)
-            copySharedObjects(sharedObjectsPath)
+            Try {
+                copyInitScript(initScriptPath)
+                copySharedObjects(sharedObjectsPath)
+            } match {
+                case scala.util.Success(_)         =>
+                    logInfo("GDAL environment prepared successfully.")
+                case scala.util.Failure(exception) =>
+                    logError("GDAL environment preparation failed.", exception)
+                    throw exception
+            }
         }
     }
 
     def enableGDAL(spark: SparkSession): Unit = {
         if (!wasEnabled(spark) && !isEnabled) {
-            isEnabled = true
-            loadSharedObjects()
-            gdal.AllRegister()
-            spark.conf.set(GDAL_ENABLED, "true")
+            Try {
+                isEnabled = true
+                loadSharedObjects()
```

Comment:
Trying to force the error on Scala... if I (intentionally) call the following on a fresh cluster (no gdal setup) to force the Failure print condition:

```
import com.databricks.labs.mosaic.gdal.MosaicGDAL
MosaicGDAL.enableGDAL(spark)
```

I get UnsatisfiedLinkError: `Can't load library: /usr/lib/libgdal.so.30` -- this is because `loadSharedObject()` call is failing before the Try/Catch. Can the logic be adjusted more to throw the failure case `logError` when loadSharedObjects fails?

Reply:
I dont think so, we can try with "java try" instead of Scala Try.
Chances are that this is too deep in JVM and just skips the exception handling.

Reply:
Most likely the loadSharedObjects call is just too deep to handle. We are going to hold with the current experience for now.